### PR TITLE
Bump opentracingVersion to latest

### PR DESCRIPTION
**Pull request by black-adder**

### CODE REVIEW

File: jaeger-core/src/test/java/com/uber/jaeger/SpanTest.java [hunk 1]
```diff
@@ -21,9 +21,7 @@
  */
 package com.uber.jaeger;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (black-adder) [inline jaeger-core/src/test/java/com/uber/jaeger/SpanTest.java]: interesting, I didn't do this manually.

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java [hunk 2]
```diff
@@ -267,19 +267,48 @@ private Span setTagAsObject(String key, Object value) {
...
+
+  @Override
+  public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+    synchronized (this) {
+      if (context.isSampled()) {
+        if (logs == null) {
+          this.logs = new ArrayList<>();
+        }
+        for (Map.Entry<String, ?> kv : fields.entrySet()) {
+          logs.add(new LogData(timestampMicroseconds, kv.getKey(), kv.getValue()));
```
Comment (black-adder) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: The intended use for jaeger log fields were that each log can be associated with multiple tags. Here, I'm creating a new log for each tag. In the JaegerThriftConverter I can add a special case for when the value is a map and generate multiple tags for the same log.

Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: -1. A log record is a timestamp + a set of fields.

File: jaeger-core/src/test/java/com/uber/jaeger/SpanTest.java (hunk 1 above)
Comment (yurishkuro) [inline jaeger-core/src/test/java/com/uber/jaeger/SpanTest.java]: you need to fix your IDE formatter. I think readme has links to IntelliJ settings

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java (hunk 2 above)
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: also, we don't want to do the conversion of fields here, we just want to stash the map somewhere and only do conversion when the span is reported.

File: build.gradle [hunk 3]
```diff
@@ -8,7 +8,7 @@ plugins {
 }
 
 
-ext.opentracingVersion = '0.15.0'
+ext.opentracingVersion = '0.20.9'
```
Comment (yurishkuro) [inline build.gradle]: there may be 0.20.10

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java [hunk 4]
```diff
@@ -267,19 +267,46 @@ private Span setTagAsObject(String key, Object value) {
...
+  }
+
+  @Override
+  public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+    synchronized (this) {
+      if (context.isSampled()) {
+        if (logs == null) {
+          this.logs = new ArrayList<>();
+        }
+        logs.add(new LogData(timestampMicroseconds, "", fields));
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: Self-inflicted inefficiency here, you reuse (String, Object) signature and then are forced to check for instanceof Map. Not to mention that there may be a payload that's a map.

I suggest simply having another constructor for LogData that stores `Map<String, ?>` fields proper.

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 5]
```diff
@@ -58,8 +58,12 @@
       for (LogData logData : logs) {
         Log jLog = new Log();
         jLog.setTimestamp(logData.getTime());
-        final Tag tag = buildTag(logData.getMessage(), logData.getPayload());
-        jLog.setFields(new ArrayList<Tag>() {{add(tag);}});
+        if (logData.getFields() != null) {
+          jLog.setFields(buildTags(logData.getFields()));
+        } else {
+          final Tag tag = buildTag(logData.getMessage(), logData.getPayload());
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: I think this is incorrect. Should be `"event": message, "payload": payload` (and drop either if null)

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java [hunk 6]
```diff
@@ -267,19 +267,52 @@ private Span setTagAsObject(String key, Object value) {
...
+  }
+
+  @Override
   public Span log(String message, /* @Nullable */ Object payload) {
     return log(tracer.clock().currentTimeMicros(), message, payload);
   }
 
   @Override
-  public Span log(long instantMicroseconds, String message, /* @Nullable */ Object payload) {
+  public Span log(long timestampMicroseconds, String message, /* @Nullable */ Object payload) {
```
Comment (black-adder) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: given that this function is deprecated, I'm not going to do a check here to see if the payload is a map.

Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: I wasn't suggesting it. Payload as map should be stringified as any other object.

File: jaeger-core/src/main/java/com/uber/jaeger/LogData.java [hunk 7]
```diff
@@ -21,26 +21,28 @@
...
-    return time;
-  }
-
-  public String getMessage() {
-    return message;
-  }
-
-  public Object getPayload() {
-    return payload;
+  LogData(long time, Map fields) {
```
Comment (felixbarny) [inline jaeger-core/src/main/java/com/uber/jaeger/LogData.java]: Add generics?

**Comment by yurishkuro** [comment]
@black-adder did you cover the new methods? why did coverage go down?

**Comment by black-adder** [comment]
I'm gonna go ahead and land this, the code that isnt covered wasn't covered before but I moved it one line down and all hell broke loose. Will add followup PR to cover the rest.

**Pull request merged**